src/net_processing.cpp

```diff
@@ -220,28 +223,65 @@ struct CNodeState {
     }
 };
 
+/** Provides locked access to a CNodeState */
+class CNodeStateAccessor {
+private:
+    std::shared_ptr<CNodeState> pstate;
+
+    CNodeStateAccessor() : pstate() { }
+    CNodeStateAccessor(std::shared_ptr<CNodeState>&& pstateIn) : pstate(std::move(pstateIn)) { ENTER_CRITICAL_SECTION(pstate->cs); }
```

Comment:
Hmm, I think you actually want these coming from net_processing anyway, no? I think if you make the ctor inline and pass \_\_LINE__ to CMutexLock, you'd end up seeing the source of the lock. Otherwise, you're always seeing it come from the same place I should think.

Taking that a step further, I think it would work to change CMutexLock's ctors to take default \_\_FILE__ and \_\_LINE__ args?